src/projects/lja/lja.cpp

```diff
@@ -129,19 +138,28 @@ int main(int argc, char **argv) {
                       corrected_final[2], skip, debug);
     if(first_stage == "rr")
         load = false;
+    if (trio) {
+        const auto binned = pipeline.TrioBinningPhase(logger, threads, dir, dir / "paternal_compressed.yak", dir / "maternal_compressed.yak", resolved[0], skip, debug);
+        logger.info() << "Resolving trio repeats with graph " << resolved[1] << std::endl;
+        pipeline.TrioSimplificationPhase(logger, threads, resolved[1], binned, corrected_final[0], lib,
```

Comment:
No, this is too ugly. Please to not call pipeline methods from outside main.cpp! Return graphs from TripSimplificationPhase and run polishing from here. trio can not include lja/pipeline.hpp ! Dependencies must be acyclic even if these are just hpp files. Also add some text at the end as it is after else. You can not say that you implemented a pipeline if you did not tell the user where to find the results.

Reply:
>Please to not call pipeline methods from outside main.cpp
But we do not have main.cpp, and this place is exactly main function, seems I miss something...

>Return graphs from TripSimplificationPhase and run polishing from here
>Dependencies must be acyclic even if these are just hpp files
But we do not have cyclic dependencies here. trio.cpp (not hpp!) includes lja/pipeline.hpp only because I want to have the separate trio target which use polishing stage. 

> Also add some text at the end as it is after else. You can not say that you implemented a pipeline if you did not tell the user where to find the results.
FXD

Reply:
> > Please to not call pipeline methods from outside main.cpp

> But we do not have main.cpp, and this place is exactly main function, seems I miss something...\

I mean the calls to polishing from trio.cpp

> > Return graphs from TripSimplificationPhase and run polishing from here
> > Dependencies must be acyclic even if these are just hpp files

> But we do not have cyclic dependencies here. trio.cpp (not hpp!) includes lja/pipeline.hpp only because I want to have the separate trio target which use polishing stage.

There is one! It is between folders and you cheated. You have lja and lja_pipeline in the same folder and trio in another and than linked them so that there is a cycle between folders but no cycle between modules. This is cheating. Also you have put all pipeline functions into lja_pipeline module. But you have put your pipeline function outside! And you link lja_pipeline to your module! It should be the other way around as with all other modules. This is also cheating! Actually the correct way to do that is to put all pipeline functions in their corresponding modules but it requires further refactoring. For now I suggest you create a separate folder for lja_pipeline, put your simplifyHaplo function into pipeline.hpp/cpp and also, as I suggested before, return graphs from TripSimplificationPhase and run polishing from here to avoid mixing trio logic with polishing logic.

Reply:
Seems that I miss something
>For now I suggest you create a separate folder for lja_pipeline, put your simplifyHaplo function into pipeline.hpp/cpp
I do not see a problem for a cycle between folders(not modules), but with suggested solution it will be again a cycle between pipeline and trio folders - trio.cpp will still need to include pipeline.hpp, wouldn't it?